lib/Dialect/Secret/Transforms/DistributeGeneric.cpp

```diff
@@ -229,66 +237,228 @@ struct SplitGeneric : public OpRewritePattern<GenericOp> {
     // RegionBranchOpInterface (scf.while, scf.if).
   }
 
-  void splitGenericAroundOp(GenericOp op, Operation &opToDistribute,
-                            PatternRewriter &rewriter) const {
-    assert(false && "not implemented");
+  // Adds new value to yield in a generic op. This requires replacing the entire
+  // generic op and cloning all its ops, since I can't find a way to modify the
+  // op's return type in-place.
+  //
+  // Returns the new op and the value range corresponding to the new result
+  // values of the generic.
+  std::pair<GenericOp, ValueRange> addNewYieldedValues(
+      GenericOp genericOp, ValueRange newValuesToYield,
+      PatternRewriter &rewriter) const {
+    YieldOp yieldOp = genericOp.getYieldOp();
+    LLVM_DEBUG(yieldOp.emitRemark() << "Before adding new values\n");
```

Comment:
Since you don't print just the originals, should you remove this statement? It seems like you the next statements print the resulting yields with the new types (not just the new ones?)

Reply:
Happy to remove it. I just found it really helpful to have a before/after dump for every modification I made to the IR, since it was really tricky to debug issues.